docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,90 @@
+= Git Submodule pattern
+
+== How this workflow pattern works:
+
+1. Create a "downstream" repo.
+2. Track your upstream git repo with a git submodule reference.
+3. Onboard to {ProductName}, use link:https://github.com/renovatebot/renovate[Renovate] to keep your Git submodule references up to date.
+4. In your downstream repo, apply patches in your Containerfile if you need to carry downstream patches.
+
+== Implementing this solution pattern
+=== Create a downstream repo
+
+A **downstream repo** is a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+The downstream repo will:
```

Comment:
I forgot to mention that it might be helpful to illustrate to readers how to configure the components. For example, what should their Context directory be? Their Dockerfile path? How would the Containerfile used differ from upstream?

Reply:
I added some basic examples for a [component definition](https://github.com/konflux-ci/docs/pull/120/files#diff-46f8b5b0ae481a31c84644c1fde822df6be41954ac5d4002de9444fe1b91fd08R29-R47) and [Dockerfile](https://github.com/konflux-ci/docs/pull/120/files#diff-46f8b5b0ae481a31c84644c1fde822df6be41954ac5d4002de9444fe1b91fd08R57-R70) using the git submodule contents and making basic edits. I'm not sure if this is what you had in mind